pkg/oauth/tokenrequest/request_token.go

```diff
@@ -133,7 +136,7 @@ func (o *RequestTokenOptions) WithChallengeHandlers(challengeHandlers ...challen
 // unstarted local callback server on the specified port.
 // If RequestTokenOptions.OsinConfig is nil, it will be defaulted using SetDefaultOsinConfig.
 // The caller is responsible for setting up the entire OsinConfig if the value is not nil.
-func (o *RequestTokenOptions) WithLocalCallback(handleAuthzURL AuthorizationURLHandlerFunc, localCallbackPort int) (*RequestTokenOptions, error) {
+func (o *RequestTokenOptions) WithLocalCallback(handleAuthzURL AuthorizationURLHandlerFunc, localCallbackPort int, clientID string) (*RequestTokenOptions, error) {
```

Comment:
I'd also suggest here to be consistent with `WithChallengeHandlers`; we can move `clientID` to be the first argument.